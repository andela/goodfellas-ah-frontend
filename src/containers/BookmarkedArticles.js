import React, { Component } from 'react';
import { connect } from 'react-redux';
import { userPlaceholderImage } from '../mixin';
import { Loader as Loading } from '../components/shared/Loading';
import { fetchProfile } from '../actions/profileActions';
import { getBookmarkedArticles } from '../actions/articleActions';
import BookmarkedArticleList from '../components/article/BookmarkedArticleList';

// eslint-disable-next-line react/prefer-stateless-function
class BookmarkedArticles extends Component {
  componentWillMount = () => {
    const {
      user: { userId },
      fetchProfile: fetchUserProfile,
      getBookmarkedArticles: fetchBookmarkedArticles,
    } = this.props;
    fetchUserProfile(userId);
    fetchBookmarkedArticles();
  }

  render() {
    const { profileStore, bookmarked } = this.props;
    const fullName = `${profileStore.user.firstname} ${profileStore.user.lastname}`;
    if (profileStore.loading && !bookmarked) return <Loading />;
    return (
      <div>
        <div className="profile-details">
          <img alt="profile" className="profile-image" src={profileStore.profile.image || userPlaceholderImage} />
          <h3 id="user-name" className="username">{fullName}</h3>
        </div>
        <h3 className="profile-title">Bookmarks</h3>
        <BookmarkedArticleList articles={bookmarked} />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  profileStore: state.profile,
  user: state.auth.ownProfile,
  bookmarked: state.articles.bookmarkedArticles,
});

export default connect(mapStateToProps, { fetchProfile, getBookmarkedArticles })(BookmarkedArticles);
