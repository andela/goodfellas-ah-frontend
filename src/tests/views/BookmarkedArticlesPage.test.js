import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import BookmarkedArticles from '../../views/BookmarkedArticlesPage';

let wrapped;

beforeEach(() => {
  wrapped = shallow(<BookmarkedArticles />);
});


describe('Bookmarked Articles UI', () => {
  describe('render features', () => {
    test('view should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
  });
});
