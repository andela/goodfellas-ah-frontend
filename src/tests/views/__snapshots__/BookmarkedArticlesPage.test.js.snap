// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmarked Articles UI render features view should render as expected 1`] = `
<_default
  className="profile"
>
  <Connect(BookmarkedArticles) />
</_default>
`;
